### PR TITLE
[REFACTORING] Split JmxProbe & GuiceProbes

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout`

unrelated

File: server/container/cli/src/main/java/org/apache/james/cli/probe/impl/JmxDataProbe.java [hunk 1]
```diff
@@ -27,13 +27,12 @@
 import javax.management.MalformedObjectNameException;
 
 import org.apache.james.domainlist.api.DomainListManagementMBean;
-import org.apache.james.probe.DataProbe;
 import org.apache.james.rrt.api.RecipientRewriteTableManagementMBean;
 import org.apache.james.rrt.lib.Mappings;
 import org.apache.james.user.api.UsersRepositoryManagementMBean;
 import org.apache.james.util.MDCBuilder;
 
-public class JmxDataProbe implements DataProbe, JmxProbe {
```
Comment (mbaechler) [inline server/container/cli/src/main/java/org/apache/james/cli/probe/impl/JmxDataProbe.java]: Is JmxDataProbe exposed as JMX? (I guess so).

Comment (chibenwa) [reply server/container/cli/src/main/java/org/apache/james/cli/probe/impl/JmxDataProbe.java]: No, the MBeans are, the JMX probe is just a way to query the MBean (it's defined on the CLI side thus it does not exist on the server side)

Comment (mbaechler) [reply server/container/cli/src/main/java/org/apache/james/cli/probe/impl/JmxDataProbe.java]: Ok.

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**